src/test/kotlin/com/atlassian/performance/tools/hardware/HardwareApdexAnalysisIT.kt

```diff
@@ -0,0 +1,269 @@
+import com.amazonaws.services.ec2.model.InstanceType.*
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder
+import com.atlassian.performance.tools.hardware.IntegrationTestRuntime.aws
+import com.atlassian.performance.tools.hardware.IntegrationTestRuntime.logContext
+import com.atlassian.performance.tools.hardware.IntegrationTestRuntime.taskName
+import com.atlassian.performance.tools.hardware.IntegrationTestRuntime.workspace
+import com.atlassian.performance.tools.hardware.failure.BugAwareTolerance
+import com.atlassian.performance.tools.jiraactions.api.*
+import com.atlassian.performance.tools.jirasoftwareactions.api.actions.ViewBacklogAction
+import com.atlassian.performance.tools.jvmtasks.api.TaskTimer.time
+import com.atlassian.performance.tools.lib.Apdex
+import com.atlassian.performance.tools.lib.table.GenericPlainTextTable
+import com.atlassian.performance.tools.lib.WeightedActionMetric
+import com.atlassian.performance.tools.lib.readResult
+import com.atlassian.performance.tools.lib.s3cache.S3Cache
+import com.atlassian.performance.tools.report.api.StandardTimeline
+import com.atlassian.performance.tools.report.api.result.EdibleResult
+import com.atlassian.performance.tools.report.api.result.RawCohortResult
+import com.atlassian.performance.tools.workspace.api.TestWorkspace
+import org.apache.logging.log4j.Logger
+import org.junit.Test
+import java.io.File
+import java.time.Duration
+import java.nio.file.FileSystems
+import java.nio.file.Paths
+
+
+class HardwareApdexAnalysisIT {
+
+    private val logger: Logger = logContext.getLogger(this::class.java.canonicalName)
+
+    // define the appdex sets we want to investigate
+    private val jiraInstanceTypeOfInterest: InstanceType = C518xlarge
+    private val nodeCountOfInterest: Int = 7
+
+    // The following needs to match the configuration of the test run you want to analyze
+    // Taken from extraLarge.Kt hold + ramp + flat values
+    private val testDuration: Duration = Duration.ZERO + Duration.ofSeconds(90) + Duration.ofMinutes(20)
+
+    @Test
+    fun shouldAnalyzeAppdex() {
+
+        // https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob
+        // action-metrics.jpt contain the appdex information
+        // status.txt is needed to pass TestWorkspace.readResult
+        val searchPattern = "**/$jiraInstanceTypeOfInterest/nodes/$nodeCountOfInterest/**/{action-metrics.jpt,status.txt}"
+        val cache = getWorkspaceCache(searchPattern)
+
+        // get the files
+        time("download") { cache.download() }
+        val runs = workspace.directory.toFile().walkTopDown()
+            .filter { file ->
+                matchesSearchPattern(file.absolutePath,"**/nodes/$nodeCountOfInterest/**runs/[0-9]*")
+            }
+            .filter { file ->
+                // ignore any empty run/# folders or ones that only contain a status.txt which indicates a failed run
+                file.listFiles().any{ it.name != "status.txt" }
```

Comment:
There are some cases the status.txt is generated with error message. E.g. one of the virtual user fails, you will still be able to find some action-matrics file.
I think you might want to filter out those file by checking if the content of the status.txt is "Ok" or not.
/com/atlassian/performance/tools/lib/Report.kt should have the logic already implemented.

Reply:
you actually did filter out those result inside apdexMetrics() :)